### PR TITLE
Code gen main

### PR DESCRIPTION
**Pull request by muscatd75**

### CODE REVIEW

File: src/main/cpp/mechanisms/intakeMechanism/generated/intakeMechanism.h [hunk 1]
```diff
@@ -13,32 +13,40 @@
...
-    // sensors
-    void DefineCANSensors() override;
+	DragonTalonFX* mainWheel;
+	DragonSolenoid* pushSolenoid;
 
-    // mechanisms
-    void DefineMechanisms() override;
+protected:
+	std::string m_ntName;
+	bool m_tuning = false;
```
[the remark this answers is not in the document]
Comment (witcpalek) [reply src/main/cpp/mechanisms/intakeMechanism/generated/intakeMechanism.h:59]: Dave, is this still in the generated code?

File: src/main/cpp/mechanisms/ejectMechanism/generated/ejectMechanism.h [hunk 2]
```diff
@@ -12,52 +13,44 @@
...
 
-	static CanSensorUsage *GetInstance();
+	DragonSolenoid* ejectSolenoid;
+//state* state_1;
 
-	CANSENSOR_USAGE GetUsage(std::string usageString);
-	std::string GetUsage(CANSENSOR_USAGE usate);
+protected:
+	std::string m_ntName;
+	std::string m_tuningIsEnabledStr;
```
Comment (witcpalek) [inline src/main/cpp/mechanisms/ejectMechanism/generated/ejectMechanism.h:47]: Dave - This shouldn't be in the mechanism;  States are where the tuning needs to be.   Joe

**Review by witcpalek — CHANGES_REQUESTED** [review]
I haven't looked at everything, but I don't understand why there is tuning stuff in the mechanism; only the states have things that can be tuned.

File: src/main/cpp/mechanisms/intakeMechanism/generated/intakeMechanism.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (muscatd75) [reply src/main/cpp/mechanisms/intakeMechanism/generated/intakeMechanism.h]: all the mechanism generated code is removed from the pull request

File: src/main/cpp/mechanisms/intakeMechanism/generated/intakeMechanism.h [hunk 3]
```diff
@@ -13,32 +13,40 @@
...
+	std::shared_ptr<nt::NetworkTable> m_table;
 
 private:
-    DragonTalonFX *m_motor1 = nullptr;
-    DragonTalonFX *m_motor2 = nullptr;
-    DragonCanCoder *m_cancoder = nullptr;
-    Example *m_example = nullptr;
-};
+	void CheckForTuningEnabled();
```
[the remark this answers is not in the document]
Comment (muscatd75) [reply src/main/cpp/mechanisms/intakeMechanism/generated/intakeMechanism.h]: all the mechanism generated code is removed from the pull request

**Pull request merged**